### DISCUSSION TITLE
When checking the plan in Dbeaver, the delete statement is executed

### DISCUSSION THREAD

**Discussion by kantuna89** [dbeaver/dbeaver#21827] [Q&A]
hello.

I used Dbeaver to check SQL plan in Oracle DB. However, I confirmed that the delete statement for the data in Plan_Table was executed. Is this normal behavior?
So why is the delete statement performed?
I want to know the reason.

Thank you.

**Comment by LonwoLonwo** [comment]
Hello @kantuna89 

If I understand correctly, the DELETE statement for the PLAN_TABLE is used to remove previous rows and add new rows - from the current explanation.

**Reply by LonwoLonwo** [reply]
As I can see in our code - there must be the "WHERE STATEMENT_ID=?" condition in the DELETE statement.

**Reply by kantuna89** [reply]
Thank you for answer.
As you said, the DELETE statement has "WHERE STATEMENT_ID=?" There are conditions.
But is it Dbeaver's default option for this delete statement to occur?
The delete statement did not occur when using tools such as sql developer.

**Reply by LonwoLonwo** [reply]
Yes, this is the default option. Again - for potential previous data deleting. To avoid duplicates.
Could you explain to me why you think this is a wrong decision?

**Reply by kantuna89** [reply]
Thank you for answer.

The information security team where I work detected the delete command and raised an issue with it.

Developers must obtain approval through an internal process to perform statements such as delete. The DBA then receives the command and executes it.

However, an unplanned delete statement was detected and this became a problem.

Is it possible to change it so that the delete command is not performed?

If changes are not possible, we will use the Read Only account to check the execution plan using SYS's PLAN_TABLE(plan_table$).

This is a global temporary table, so if the session ends, all contents will be lost and no delete commands are expected to be issued.

**Reply by LonwoLonwo** [reply]
You can create a bug ticket for this case.
